core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/Envelope.java

```diff
@@ -186,7 +186,6 @@ public double maxSpeedInRange(double beginPos, double endPos) {
     }
 
     private long interpolateUS(double position, boolean isArrivalAt) {
-        assert continuous : "interpolating times on a non continuous envelope is a risky business";
```

Comment:
Why?

Reply:
Interpolating the *speed* would have been risky, but we're interpolating the *time*, which is continuous. It's not differentiable but we don't care about that. 

That assert in this specific method never actually made sense, we just haven't had a need to interpolate times on a non continuous envelope yet. But we do now, we use the MRSP to estimate a fastest time.